BookWorm/BookWorm/View Controllers/SearchViewController.m

```diff
@@ -27,7 +29,90 @@ - (void)viewDidLoad {
     [super viewDidLoad];
     self.tableView.dataSource = self;
     self.arrayOfBooks = [[NSMutableArray alloc] init];
+    self.autocompleteTitles = [[NSMutableArray alloc] init];
+    self.searchTitle.delegate = self;
     // Do any additional setup after loading the view.
+    
+    self.titleTableView.dataSource = self;
+    self.titleTableView.scrollEnabled = YES;
+    self.titleTableView.hidden = YES;
+}
+
+- (IBAction)checkMarkPressed:(id)sender {
+    self.titleTableView.hidden = YES;
+}
+
+- (BOOL)textField:(UITextField *)textField
+    shouldChangeCharactersInRange:(NSRange)range
+    replacementString:(NSString *)string {
+     
+    if (textField == self.searchTitle) {
+        NSString *substring = [NSString stringWithString:self.searchTitle.text];
+        substring = [substring
+          stringByReplacingCharactersInRange:range withString:string];
+        [self searchAutocompleteEntriesWithSubstring:substring];
+    }
+    
+  return YES;
+}
+
+- (void)searchAutocompleteEntriesWithSubstring:(NSString *)substring {
+    [self.autocompleteTitles removeAllObjects];
+    
+    NSString *baseURL = @"https://www.googleapis.com/books/v1/volumes?q=";
+    NSString *titleString;
+    if ([self.searchTitle.text length] != 0) {
+        titleString = [baseURL stringByAppendingFormat:@"%@%@", @"+intitle:", [self splitSearchWords:self.searchTitle.text]];
+    }
+    
+    NSString *path = [[NSBundle mainBundle] pathForResource: @"Keys" ofType: @"plist"];
+    NSDictionary *dict = [NSDictionary dictionaryWithContentsOfFile: path];
+    NSString *googleKey = [dict objectForKey: @"GoogleBooksAPIKey"];
+    NSString *finalURL = [titleString stringByAppendingFormat:@"%@%@", @"&key=", googleKey];
+    NSURL *url = [NSURL URLWithString:finalURL];
+    NSLog(@"Final URL");
+    NSLog(@"%@", finalURL);
+    NSURLRequest *request = [NSURLRequest requestWithURL:url cachePolicy:NSURLRequestReloadIgnoringLocalCacheData timeoutInterval:10.0];
+    NSURLSession *session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration] delegate:nil delegateQueue:[NSOperationQueue mainQueue]];
+    NSURLSessionDataTask *task = [session dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+           if (error != nil) {
+               NSLog(@"%@", [error localizedDescription]);
```

Comment:
You can short-circuit the error, so the rest of this code block isn't as nested.